@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -201,18 +208,74 @@ export class TileDataSource<TileType extends Tile> extends DataSource {
     getTile(tileKey: TileKey): TileType | undefined {
         const tile = this.m_tileFactory.create(this, tileKey);
 
-        tile.tileLoader = new TileLoader(
-            this,
-            tileKey,
-            this.m_options.dataProvider,
-            this.decoder,
-            0
-        );
+        const mortonCode = tileKey.mortonCode();
+        const tileLoader = this.m_tileLoaderCache.get(mortonCode);
+        // Filter out disposed TileLoaders.
+        if (tileLoader !== undefined && tileLoader.state !== TileLoaderState.Disposed) {
+            tile.tileLoader = tileLoader;
+        } else {
+            const newTileLoader = new TileLoader(
+                this,
+                tileKey,
+                this.m_options.dataProvider,
+                this.decoder,
+                0
+            );
+            tile.tileLoader = newTileLoader;
+            // We don't cache tiles with level 4 and above, at this level, there are 16 (2^4)
+            // tiles horizontally, given the assumption that the zoom level assumes the tile
```

Comment:
This assumption depends on the storageLevelOffset of the datasource, so you should add it to the condition below.

Reply:
I will reword.